### DISCUSSION TITLE
Pyarmor v8 Follow Imports

### DISCUSSION THREAD

**Discussion by grinchify** [dashingsoft/pyarmor#1419] [Q&A] [answered]
Running the pyarmor command pyarmor gen  main.py

and main.py has an import from utilities.utils import * and import requests

however seems like pyarmor does not follow the imports?

**Comment by grinchify** [comment]
```
from test import run

print("Starting")
run()

```
```
def run():
    print("Import Success")
```

pyarmor gen main.py

Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen main>", line 1, in <module>
ImportError: cannot import name 'run' from 'test' (C:\Program Files\Python39\lib\test\__init__.py)

**Reply by Warfront1** [reply]
If you could put together a sample repository, I could probably help you.
However, when I use the `pyarmor gen` command I typically like to place my python scripts into a folder.
`python -m pyarmor.cli gen -O obfdist src/`
In the above command I am generating obfuscated scripts for all python scripts found in `src` including scripts inside of deeper folder structures.

**Reply by grinchify** [reply]
Ah I see, that works well. Any chance you got discord or email?

**Reply by Warfront1** [reply]
You can reach me on discord via my username `warfront1`.

**Comment by grinchify** [comment, marked as answer]
https://pyarmor.readthedocs.io/en/latest/topic/rftmode.html
